lib/yard/cli/yardoc.rb

```diff
@@ -273,6 +277,8 @@ def run(*args)
           end
         end
 
+        exit 1 if fail_on_warning && log.warned
```

Comment:
You can use `abort` instead of `exit 1` - the (slight) benefit `abort` has is that the provided failing exit status code is something which works for the current platform.